src/jit.rs

```diff
@@ -755,78 +801,82 @@ impl<'a> JitMemory<'a> {
+                    emit_call(self, func_ptr as i64);
+
+                    // Throw error if the result indicates one
+                    emit_load(self, OperandSize::S64, RDI, map_register(0), 0);
+                    emit_jcc(self, 0x85, TARGET_PC_EPILOGUE);
+
+                    // Store Ok value in result register
+                    emit_load(self, OperandSize::S64, RDI, map_register(0), 8);
                 },
                 ebpf::CALL_REG  => { unimplemented!() },
-                ebpf::EXIT       => {
+                ebpf::EXIT      => {
                     if insn_ptr != prog.len() / ebpf::INSN_SIZE - 1 {
                         emit_jmp(self, TARGET_PC_EXIT);
                     };
                 },
 
-                _                => return Err(JITError::UnknownOpCode(insn.opc, insn_ptr)),
+                _               => return Err(JITError::UnknownOpCode(insn.opc, insn_ptr)),
             }
 
             insn_ptr += 1;
         }
 
-        // Epilogue
+        // Quit gracefully
         set_anchor(self, TARGET_PC_EXIT);
 
-        // Move register 0 into rax
-        if map_register(0) != RAX {
-            emit_mov(self, map_register(0), RAX);
-        }
+        // Store result in optional type
+        emit_store(self, OperandSize::S64, map_register(0), RDI, 8);
+        // Also store that no error occured
+        emit_load_imm(self, map_register(0), 0);
+        emit_store(self, OperandSize::S64, map_register(0), RDI, 0);
+
+        // Epilogue
+        set_anchor(self, TARGET_PC_EPILOGUE);
 
         // Deallocate stack space
         emit_alu64_imm32(self, 0x81, 0, RSP, CALL_FRAME_SIZE as i32);
 
-        emit_pop(self, R15);
-        emit_pop(self, R14);
-        emit_pop(self, R13);
-        emit_pop(self, RBX);
-        emit_pop(self, RBP);
+        // Restore registers
+        for reg in CALLEE_SAVED_REGISTERS.iter().rev() {
+            emit_pop(self, *reg);
+        }
 
         emit1(self, 0xc3); // ret
 
         // Division by zero handler
         set_anchor(self, TARGET_PC_DIV_BY_ZERO);
-        fn log(pc: u64) -> i64 {
-            // Write error message on stderr.
-            // We would like to panic!() instead (but does not work here), or maybe return an
-            // error, that is, if we also turn all other panics into errors someday.
-            // Note: needs `use std::io::Write;`
-            //     let res = writeln!(&mut std::io::stderr(),
-            //                        "[JIT] Error: division by zero (insn {:?})\n", pc);
-            //     match res {
-            //         Ok(_)  => 0,
-            //         Err(_) => -1
-            //     }
-            pc as i64 // Just to prevent warnings
-        };
         emit_mov(self, RCX, RDI); // muldivmod stored pc in RCX
-        emit_call(self, log as usize);
-        emit_load_imm(self, map_register(0), -1);
-        emit_jmp(self, TARGET_PC_EXIT);
+        // TODO: Store error
```

Comment:
Isn't the result stored above (lines 856 to 863)?

Reply:
No this is for divide by zero exceptions, the one above you referenced is for syscalls.